sections/introduction.tex

```diff
@@ -1,10 +1,17 @@
+ 
 
-Imagine a classroom with students following a presentation which is concluded with a vote from students' smart phones. Although students are connected to the same WiFi, such votes are usually performed using online voting platforms where all voters have to connect to the Internet for accessing the platform. Despite the obvious unnecessary traffic, this scenario has another significant problem: the voting cannot be performed upon failing Internet access. This seems almost abstruse knowing that all participants are connected to the same local network.
+The Internet as a platform of communication has developed a one-and-only character in the previous decade: no matter where devices and users are located, they connect to the Internet and talk to each other through it. Even in scenarios where these devices are close to each other on the same network, local architecture is rarely leveraged and all traffic has to go through the global network, possibly traversing multiple autonomous systems throughout the globe. Driven in particular by the growing domain of the Internet of things (IoT) where different ``smart" local devices require network communication, a movement towards making more use of local network infrastructure seems just as reasonable. Nonetheless, adding and configuring these devices in a local-area network may be error prone, time consuming, and non-inviting to the public that purchase them. On top of that, applications run by these devices may require unnecessary Internet connection in order to communicate with their peers. 
 
-The Mozilla foundation was one of the first major contributors that tackled this problem with their Firefox extension FlyWeb. The new technology allows clients of Web applications to start their own local Web server from within the browser. Advertised in the local network through MDNS, other devices in the network can detect the new server and can connect to it via HTTP or WebSocket. This technique offers a range of new possibilities for Web applications and local vs. global network behavior. But there are also severe limitations, one of them being the complete lack of fault tolerance: when the server dies, the client-server network dies with it. Since this technology is inherently driven by the idea of any device being able to become a server, we assume that it is more likely that servers misbehave in comparison to the traditional scenario of "real" Web servers. We therefore regard fault tolerance as very important in such networks and we think the lack of mechanisms for graceful recovery is problematic.
```

Comment:
I added a reference to zero-configuration networks because they were the first ones to address device configuration and service discovery in a local network. FlyWeb is built on top of it so it is interesting that we reference zeroconf prior to flyweb